src/engine/source/builder/src/builders/opmap/opBuilderHelperMap.cpp

```diff
@@ -2012,7 +2046,338 @@ MapOp opBuilderHelperIPVersionFromIPStr(const std::vector<OpArg>& opArgs,
+                                          sportOutOfRangeTransportTrace,
+                                          missingTrace(sportRef),
+                                          numberTrace(sportRef),
+                                          sport))
+            {
+                RETURN_FAILURE(runState, json::Json {}, *err);
+            }
+
+            if (const auto err = readPort(dportRef,
+                                          true,
+                                          PORT_MAX,
+                                          dportOutOfRangeTransportTrace,
+                                          missingTrace(dportRef),
+                                          numberTrace(dportRef),
+                                          dport))
+            {
+                RETURN_FAILURE(runState, json::Json {}, *err);
+            }
+        }
+        else
+        {
+            // Otros: opcional (fallback 0/0), rango 0..65535
+            if (const auto err = readPort(sportRef,
+                                          false,
+                                          PORT_MAX,
+                                          sportOutOfRangeTransportTrace,
+                                          missingTrace(sportRef),
+                                          numberTrace(sportRef),
+                                          sport))
+            {
+                RETURN_FAILURE(runState, json::Json {}, *err);
+            }
+
+            if (const auto err = readPort(dportRef,
+                                          false,
+                                          PORT_MAX,
+                                          dportOutOfRangeTransportTrace,
+                                          missingTrace(dportRef),
+                                          numberTrace(dportRef),
+                                          dport))
+            {
+                RETURN_FAILURE(runState, json::Json {}, *err);
+            }
+        }
+
+        const auto cid = base::utils::CommunityId::getCommunityIdV1(saddr, daddr, sport, dport, proto, defaultSeed);
+
+        if (std::holds_alternative<base::utils::CommunityId::CommunityError>(cid))
+        {
+            switch (std::get<base::utils::CommunityId::CommunityError>(cid))
+            {
+                case base::utils::CommunityId::CommunityError::BuildBufferFailed:
+                    RETURN_FAILURE(runState, json::Json {}, buildBufferFailureTrace);
+                case base::utils::CommunityId::CommunityError::Sha1Failure:
+                    RETURN_FAILURE(runState, json::Json {}, sha1FailureTrace);
+                case base::utils::CommunityId::CommunityError::Base64Failure:
+                    RETURN_FAILURE(runState, json::Json {}, base64FailureTrace);
+                case base::utils::CommunityId::CommunityError::Unknown:
+                default: RETURN_FAILURE(runState, json::Json {}, computeFailureTrace);
+            }
+        }
```

Comment:
This can be solved using the following declaration: `using RespOrError = std::variant<T, Error>;` ///< T and Error are defined in error.hpp